addon/-private/system/store.js

```diff
@@ -1093,7 +1137,10 @@ Store = Service.extend({
     assert("You tried to load a query but you have no adapter (for " + typeClass + ")", adapter);
     assert("You tried to load a query but your adapter does not implement `query`", typeof adapter.query === 'function');
 
-    return promiseArray(_query(adapter, this, typeClass, query, array));
+    let pA = promiseArray(_query(adapter, this, typeClass, query, array));
+
+    pA.finally(() => { heimdall.stop(token); });
+    return pA;
```

Comment:
This should be wrapped in `instrument` so that we don't have an extra `finally` dangling around.